racksdb/web/ui/src/views/DatacenterRoomView.vue

```diff
@@ -108,57 +146,90 @@ const props = defineProps({
+
+  <div class="flex justify-center py-3">
+    <div class="flex justify-between min-w-[75vw]">
+      <div @click="sortRacks()" class="flex items-center pt-5">
+        <div v-if="alphabeticalOrder">
+          <BarsArrowDownIcon class="h-7 w-7" />
+        </div>
+        <div v-else>
+          <BarsArrowUpIcon class="h-7 w-7" />
+        </div>
+        <p>Sort racks</p>
       </div>
+
+      <SwitchGroup>
+        <div class="pt-5 flex items-center">
+          <SwitchLabel class="mx-4">Hide empty racks</SwitchLabel>
+          <Switch
+            v-model="hideEmpty"
+            :class="hideEmpty ? 'bg-purple-900' : 'bg-purple-700'"
+            class="relative inline-flex h-[28px] w-[64px] shrink-0 cursor-pointer rounded-full border-2 border-transparent transition-colors duration-200 ease-in-out focus:outline-none focus-visible:ring-2 focus-visible:ring-white/75"
+          >
+            <span
+              aria-hidden="true"
+              :class="hideEmpty ? 'translate-x-9' : 'translate-x-0'"
+              class="pointer-events-none inline-block h-[24px] w-[24px] transform rounded-full bg-white shadow-lg ring-0 transition duration-200 ease-in-out"
+            />
+          </Switch>
+        </div>
+      </SwitchGroup>
     </div>
   </div>
 
-  <div class="flex justify-center">
-    <table
-      v-if="datacenterDetails"
-      class="w-screen text-base text-center text-gray-500 dark:text-gray-400"
-    >
-      <thead class="text-lg text-gray-700 uppercase bg-gray-50 dark:bg-gray-700 dark:text-gray-400">
+  <div class="flex justify-center pb-10">
+    <table class="min-w-[75vw] text-base text-center text-gray-500 dark:text-gray-400 table-fixed">
+      <thead
+        class="text-lg text-gray-700 uppercase bg-gray-50 dark:bg-gray-700 dark:text-gray-400 h-24"
+      >
         <tr>
-          <th scope="col" class="px-6 py-3">Name</th>
-          <th scope="col" class="px-6 py-3">Fill rate</th>
-          <th scope="col" class="px-6 py-3">List of infrastructures</th>
+          <th scope="col" class="w-24">Name</th>
+          <th scope="col" class="w-24">Fill rate</th>
+          <th scope="col" class="w-24">List of infrastructures</th>
         </tr>
       </thead>
 
-      <template v-for="room in datacenterDetails.rooms" :key="room">
-        <template v-for="row in room.rows" :key="row">
-          <template v-for="rack in row.racks" :key="rack">
-            <tbody>
-              <tr>
-                <td>{{ rack.name }}</td>
-                <td>{{ (rack.fillrate * 100).toFixed(0) }}%</td>
-                <td class="capitalize">
-                  <router-link
-                    :to="{ name: 'infrastructuredetails', params: { name: 'mercury' } }"
-                    >{{ listInfrastructures(rack.name).join(' , ') }}</router-link
-                  >
-                </td>
-              </tr>
-            </tbody>
-          </template>
-        </template>
+      <template v-if="filteredRacks!.length > 0">
```

Comment:
`filteredRacks` cannot be _null_ or _undefined_ then I don't see the utility of the bang operator.